transcom/mymove
# MB-2494 Use ClientError for Prime and Support API responses

## Description

This PR changes all of the responses (except for Server Error) in the `prime.yaml` and `support.yaml` to use the `ClientError` definition. This means all errors (except Server Errors) will have, at minimum:

- title (a standard value)
- detail (the custom message)
- instance ID (UUID for the trace)

## Setup

These changes can be tested by triggering (handled) errors with any Prime or Support endpoint. More detailed instructions for how to set up access to these APIs using either the CLI or Postman can be found here: https://github.com/transcom/prime_api_deliverable/wiki/API-Endpoints

### Example

- Grab a shipment ID and eTag using the Fetch MTOs endpoint in the Prime API. 
- Using the Support Endpoint "Update MTO Shipment Status" `/mto-shipments/{mtoShipmentID}/status`, try to update its status, but change one of the character of the `mtoShipmentID`. You should see a 404 response like:

```json
{
    "detail": "id: 15d360aa-c80a-4fca-9502-8540fd28b138 not found ",
    "instance": "7c2e334b-bc4a-4e49-bb17-a8e43ad7e1a4",
    "title": "Not Found Error"
}
```

- Correct the typo'ed UUID. Now try to submit a body with the status set to REJECT, but don't pass in a value for `rejectionReason`. You should see a 422 response like:

```json
{
    "detail": "The input provided did not pass validation.",
    "instance": "933cd3eb-2c01-43ea-87ca-35aff6dcf464",
    "title": "Validation Error",
    "invalidFields": {
        "rejection_reason": [
            "RejectionReason can not be blank."
        ]
    }
}
```

- Now mark it as APPROVED and submit. You should get a successful payload. Grab the new eTag value and put it in the header.

- Submit it again with a new status (SUBMITTED/REJECTED) or the same one. You should see a 409 response like:

```json
{
    "detail": "shipment with id 'a0cf3769-3419-46ad-b765-6d95f108f732' can not transition status from 'APPROVED' to 'REJECTED'. Must be in status 'SUBMITTED'.",
    "instance": "06c67c54-47b2-403a-8b06-946c0f591966",
    "title": "Conflict Error"
}
```

- Repeat the same kinds of tests for some other endpoint in the Prime/Support APIs

**NOTE:** The Create MTO Shipment and Create Upload endpoints do not have any error payloads. Those will be added in two future tickets, so those endpoints would not be good ones to test with for this PR (since this one was for converting the existing errors to the new format)

## Code Review Verification Steps


## References

* [Jira story](https://dp3.atlassian.net/browse/MB-2494) for this change
